src/authenticator.rs

```diff
@@ -1335,6 +1347,61 @@ where
         }
         return Ok(());
     }
+
+    fn yk_hmac<const R: usize>(&mut self, req: YKGetHMAC, reply: &mut Data<{ R }>) -> Result {
+        // Get HMAC slot command
+
+        let credential = self
+            .load_credential(req.get_credential_label()?)
+            .ok_or(Status::NotFound)?;
+        let key: &[u8] = &credential.secret;
+
+        // Make sure the set Credential is the right kind
+        ensure(
+            credential.kind == Kind::Hmac,
+            Status::IncorrectDataParameter,
+        )?;
+
+        let signature = hmac_challenge(&mut self.trussed, Algorithm::Sha1, req.challenge, key)?;
```

Comment:
I think we should check the algorithm of the credential, and remove the check bellow for the length. I don't really understand why its here (maybe for debug? In that case it could stay as a `debug_assert`).

Reply:
Exactly. As noted earlier.